### ISSUE TITLE
Player ghost ban from the app

### ISSUE THREAD

**Issue by zelytra** [zelytra/BetterFleet#314]
## Summary

"Sometimes" when a player is using normaly the application it can occure that it is in a session and then be kicked for no reason and cannot longer connect/create any session until the app restart

## Steps to reproduce

No repduction for now, still investigating

## What is the current bug behavior?

Player can no longer join/create a session

## What is the expected correct behavior?

Player is not locked in deadloop

## Relevant logs and/or screenshots

```
2024-07-06 19:35:04,459 INFO  [fr.zel.ses.SessionSocket] (vert.x-eventloop-thread-1) [player1] Connected !
2024-07-06 19:35:04,459 INFO  [fr.zel.ses.SessionManager] (vert.x-eventloop-thread-1) [363A07D] player1 Leave the session !
2024-07-06 19:35:04,459 INFO  [fr.zel.ses.SessionManager] (vert.x-eventloop-thread-1) [363A07D] Has been disbanded
2024-07-06 19:35:04,459 WARN  [fr.zel.ses.SessionSocket] (vert.x-eventloop-thread-1) [UNDEFINED PLAYER] Disconnected
2024-07-06 19:35:04,459 INFO  [fr.zel.ses.SessionManager] (vert.x-eventloop-thread-1) [A2C5152] player1 Join the session !
2024-07-06 19:35:04,459 ERROR [fr.zel.ses.SessionManager] (vert.x-eventloop-thread-1) Unable to send message: io.netty.channel.StacklessClosedChannelException
2024-07-06 19:36:11,226 INFO  [fr.zel.ses.soc.sec.SocketSecurityEndpoints] (executor-thread-117) [GET] /socket/register
2024-07-06 19:36:11,472 INFO  [fr.zel.ses.SessionSocket] (vert.x-eventloop-thread-4) [ANYONE] Connecting...
2024-07-06 19:36:11,499 INFO  [fr.zel.ses.SessionSocket] (vert.x-eventloop-thread-4) [player1] Connected !
2024-07-06 19:36:11,499 INFO  [fr.zel.ses.SessionManager] (vert.x-eventloop-thread-4) [0E87A77] Session created !
2024-07-06 19:36:11,499 INFO  [fr.zel.ses.SessionManager] (vert.x-eventloop-thread-4) [A2C5152] player1 Leave the session !
2024-07-06 19:36:11,499 WARN  [fr.zel.ses.SessionSocket] (vert.x-eventloop-thread-4) [UNDEFINED PLAYER] Disconnected
2024-07-06 19:36:11,499 INFO  [fr.zel.ses.SessionManager] (vert.x-eventloop-thread-4) [0E87A77] player1 Join the session !
2024-07-06 19:36:11,505 ERROR [fr.zel.ses.SessionManager] (vert.x-eventloop-thread-4) Unable to send message: io.netty.channel.StacklessClosedChannelException
```

## Possible fixes

Right a Unit Test to reproduce this bug case. The problem come only from the backend but no clue how.

**Comment by zelytra** [comment]
Report of the player X1:
## Description:
Since the latest update, BetterFleet v0.2.3 has been experiencing significant issues. A user reported their username appearing twice in the same session, one instance in a server even when they were in the menu.

## Steps to Reproduce:

    Open BetterFleet.
    Username appears twice in the same session.
    Close and reopen BetterFleet.
    Click "Create a session" - no response.
    Click "Join a session" and enter the code of an existing session - no response.

## Additional Details:

    The user has been using the application since the alpha version.
    The issue started on 06/27/2024 at 12:01.
    The user promoted another person in the session, was demoted back, left the session, and returned. This resolved the duplicate username issue, but the server with the duplicate username remained visible to others.
    Multiple restarts and updates (Windows, Microsoft Store) did not resolve the issue.

Temporary Fix Attempted:

    Restarting the application to unblock the user.

Investigation Needed:

    Cause of the duplicated username.
    Resolving the unresponsive "Create a session" and "Join a session" actions.

**Comment by zelytra** [comment]
Report of player X2: 
## Description:
A user reported issues when attempting to create or join a server in BetterFleet. Clicking to create a server does nothing, and when joining a server with a code, friends can see the user, but the user sees nothing.

## Steps to Reproduce:

    Install BetterFleet.
    Log in to the application.
    Click to create a server – no response.
    Enter the code to join a server – friends can see the user, but the user sees nothing.

## Additional Details:

    The issue may be related to the difference between the Microsoft account used for the game and the one on the user's PC, although this was dismissed by support.
    The problem occurs immediately after a fresh install without any previous successful server creation or joining attempts.

## Temporary Fix Attempted:

    Restarting the BetterFleet server temporarily resolves the issue.

## Investigation Needed:

    Determine the root cause of the unresponsive server creation and visibility issues when joining a server.
    Investigate the initial state leading to this problem immediately after installation.

## User Offer:

    The user is willing to assist further and can provide exact actions taken if the issue reoccurs.

**Comment by zelytra** [comment]
Maybe a investigation need to be executed arround player session managments

**Comment by zelytra** [comment]
The reproduction case is :
`One player create a session, another instance of te same player try to create a session also. The player is dead lock`

The problem can appear in a case were the player arrive to create 2 websocket, with the same instance (rare case)

Related to the fix #259, but the fix was concerning only SessionManager and missed a level of test in SessionSocket.

Test case has been writted waiting to fix.
